Terminal.Gui/ConsoleDrivers/NetDriver/NetEvents.cs

```diff
@@ -58,32 +60,6 @@ private ConsoleKeyInfo ReadConsoleKeyInfo (CancellationToken cancellationToken,
                 return Console.ReadKey (intercept);
             }
 
-            if (AnsiEscapeSequenceRequestUtils.IncompleteCkInfos is null && EscSeqRequests is { Statuses.Count: > 0 })
-            {
-                if (_retries > 1)
-                {
-                    if (EscSeqRequests.Statuses.TryPeek (out AnsiEscapeSequenceRequestStatus? seqReqStatus) && string.IsNullOrEmpty (seqReqStatus.AnsiRequest.AnsiEscapeSequenceResponse?.Response))
-                    {
-                        lock (seqReqStatus.AnsiRequest._responseLock)
-                        {
-                            EscSeqRequests.Statuses.TryDequeue (out _);
-
-                            seqReqStatus.AnsiRequest.RaiseResponseFromInput (null);
-                        }
-                    }
-
-                    _retries = 0;
-                }
-                else
-                {
-                    _retries++;
-                }
-            }
-            else
-            {
-                _retries = 0;
-            }
-
```

Comment:
Hangs because you removed this code that returns if it's an invalid terminator.

Reply:
> Hangs because you removed this code that returns if it's an invalid terminator.

You still get invalid terminators being picked up:

![image](https://github.com/user-attachments/assets/791e89ff-579a-40d2-bf31-df4520861481)

The issue arises when you never get your terminator at all.  i.e. you get lots of ANSI requests but none of them match your terminator.

In this case we need the loop to detect stale requests and give up i.e. call `seqReqStatus.AnsiRequest.RaiseResponseFromInput (null);`

Reply:
I think my behavior is correct. The user really wanted to send the `\esc[0c` but made a mistake on the terminator. Then the driver return the right response with an error that the user didn't use the correct terminator. This is a better information than only return null. It return false anyway but with the correct response.

Reply:
Sorry you have misunderstood, I agree with this approach.  This screenshot is from this branch

The null is when no response comes in right? I.e. for a while

In which case we need a way to raise that in the new code in this PR like you had it before just not with that loop.

I.e. working with the parser rather than against it.

I had a whole scheduler class that did this exact job but probably in the blocking case we can find a simpler couple of lines like recording time of request and unblocking if > x ms

Reply:
I test now your branch but it still block when I debug with WSL Linux. I've to first resolve this issue. I think `BlockingCollection` isn't working as expected like in Windows.

Reply:
>  I think BlockingCollection isn't working as expected like in Windows.

That seems unlikely, its a core dotnet class.

What point in your code is it blocking in if you Attach Debugger?

![image](https://github.com/user-attachments/assets/0781e38a-fb90-4484-a8d3-b1b1c01237a8)

![image](https://github.com/user-attachments/assets/00430873-9994-4003-b4c3-a9bed9d51ff7)

Reply:
Its after I attached and move the mouse on the app.

Reply:
Ah I can reproduce this, its only when the debugger is attached during running right?

Reply:
It does not happen in `v2_develop` or `ansi-parser` branches.  `ansi-parser` is using BlockingCollection so can rule that out.

Reply:
I think the problem is when there are input main loop try to process it before the enqueued returned.

Reply:
The only information I get when the app in `Linux` in this in the output window, after disabling `Just My Code`.

`The thread '.NET TP Worker' (522) has exited with code 0 (0x0).`

Reply:
Another detail is that this only hang in Linux with mouse events and not with key events.

Reply:
> Hangs because you removed this code that returns if it's an invalid terminator.

Fixed in:

[ba51251](https://github.com/BDisp/Terminal.Gui/pull/207/commits/ba51251d733f70512cfa51a2f9ea5b900b7b1e7a)

If request does not get answered within 200 ms it will get abandonned and raised as `null`

Reply:
Thanks. But now I'm focused on why it's hanging in `Linux`. That is worry me a lot.

Reply:
You can get it to break in debugger if you do not give focus to the Terminal.Gui yet.

e.g. put break point in ReadConsoleKeyInfo

The input loop happily iterates until you give a click or something - then deadlock `ProcessInputQueue()` no longer loops.

Reply:
I already the bug and its working fine now. Now I only am find why in `Linux` I have `IncompleteCkInfos` and its not filling well. So, each job must have his own thread. I've to add again the slims and now it's ok. Beside it's now working in Windows we have to guarantee that will work well on Linux as well.

Reply:
Ok please prioritize merging this, we do not need IncompleteCkInfos if using AnsiParser as it already handles that

Reply:
Please understand that I also am handling `IncompleteCkInfos` well in `CursesDriver`. In `NetDriver` `IncompleteCkInfos` is only happen on `Linux` and I want to understand why it's happening. I prefer You maintain this branch always merged with my PR. Thus we can find any others issues.

Reply:
But there are already conflicts. If we can agree this works for net driver get it merged and I will implement same hook-ins for windows and Unix. I already implemented for all drivers in ansi-parser branch so we know it's possible.

You keep making changes and want me to integrate over and over - it is a lot of extra work for me for no reason I can see.

Reply:
But the drivers as you have doesn't work well on Linux and I'm afraid to mess-up all again. I also will revert the `WindowsDriver` which I think it's causing git actions failure. I beg you please just a little of patient.

Reply:
> But the drivers as you have doesn't work well on Linux

On this branch or ansi-parser branch? I understand ansi-parser has this issue.

But this branch is a rewrite to fit in with the blocking approach.  Can you elaborate on what is not working. Maybe we can fix together?

> please just a little of patient

 Ok sure

Reply:
I already pointed what it's wrong with your implementation of `BlockingCollection` in this comment https://github.com/gui-cs/Terminal.Gui/pull/3791#discussion_r1835433290. I hope you understand the big problem here, mainly with the Linux. Without running timers nor handlers nor inputs the CPU is a very high supercharge about of 13%. With the fixes I implemented in my PR, running the scenario ansi with 20 requests I have 6%, which is normal because it's running timers, inputs and redraw.

Reply:
CPU is fine in this branch from what I can see? I reverted the changes to `_eventReady` I had made in my https://github.com/gui-cs/Terminal.Gui/pull/3791 branch so now methods are all blocking while idle (and no idle timers etc).